doc/ext/traffic-server.py

```diff
@@ -371,6 +371,26 @@ def xref_cleanup(app, env, node, contnode):
                 return node
     return;
 
+def make_github_link(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    """
+    This docutils role lets us link to source code via the handy :ts:git: markup.
+    Link references are rooted at the top level source directory. All links resolve
+    to GitHub.
+
+    Examples:
+
+        To link to proxy/Main.cc:
+
+            Hi, here is a link to the proxy entry point: :ts:git:`proxy/Main.cc`.
+
+        To link to CONTRIBUTING.md:
+
+            If you want to contribute, take a look at :ts:git:`CONTRIBUTING.md`.
+    """
+    url = 'https://github.com/apache/trafficserver/blob/master/{}'
```

Comment:
I think it wouldn't work well because we provide old documentations from the branches. If files referenced with this macro was moved on master, old documentations would have dead links.

Can we reference files on the branch currently checked out somehow?